src/main/java/data/TaskManager.java

```diff
@@ -109,22 +143,76 @@ public static void addManager(Scanner scanner, WeekView weekView, boolean inMont
             checkIfDateInCurrentWeek(date, weekView);
         }
 
+        // vvv Below methods should be recreated when console inputs are streamlined
+        System.out.println("Enter the type of task (T for Todo, E for event, D for deadline):");
+        TaskType taskType = parseTaskType(scanner.nextLine().trim().toUpperCase());
+        // ^^^ Above methods should be recreated  when console inputs are streamlined
+
         System.out.println("Enter the task description:");
         String taskDescription = scanner.nextLine().trim();
 
-        addTask(date, taskDescription);
+        // vvv Below methods should be recreated when console inputs are streamlined
+        if (taskType == DEADLINE) {
+            System.out.println("Enter the deadline of this task:");
+            String[] deadline = new String[]{scanner.nextLine().trim()};
+
+            addTask(date, taskDescription, taskType, deadline);
+        } else if (taskType == EVENT) {
+            System.out.println("Enter the start date of this task:");
+            String startDate = scanner.nextLine().trim();
+            System.out.println("Enter the end date of this task:");
+            String endDate = scanner.nextLine().trim();
+            String [] startAndEndDates = new String[]{startDate, endDate};
+
+            addTask(date, taskDescription, taskType, startAndEndDates);
+        } else {
+            String[] dummyDates = {null}; // dummy String array to pass into function call
+            addTask(date, taskDescription, taskType, dummyDates);
+        }
+        // ^^^ Above methods should be recreated  when console inputs are streamlined
+
         saveTasksToFile(tasks, Storage.FILE_PATH); // Updates tasks from hashmap into tasks.txt file
         System.out.println("Task added.");
     }
 
     /**
-     * Prompts user for updated task description
+     * Method that parses the TaskType to be specified based on the user's input.
      *
-     * @param scanner User input
-     * @param weekView Current week being viewed
-     * @param inMonthView Whether month is being viewed
-     * @param taskManager The taskManager class being used
-     * @throws TaskManagerException Throws exception when not in correct week/month view
+     * @param userInput The String containing the user's cleaned input.
+     * @return TaskType of the user's choosing.
+     */
+    public static TaskType parseTaskType(String userInput) {
+        TaskType currentTaskType;
+        switch (userInput) {
+        case "T":
+            currentTaskType = TODO;
+            break;
+
+        case "D":
+            currentTaskType = DEADLINE;
+            break;
+
+        case "E":
+            currentTaskType = EVENT;
+            break;
+
+        default:
+            currentTaskType = null;
+            // intentional fallthrough since addTask method checks for invalid taskType already
+        }
+
+        return currentTaskType;
+    }
```

Comment:
Great separation of the parsing logic and the user interface printing for better readability!